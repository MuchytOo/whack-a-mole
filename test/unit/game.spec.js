import { shallowMount } from '@vue/test-utils'
import GamePage from '../../pages/gamePage'

jest.mock('vue-router')
jest.useFakeTimers()

describe('GamePage.vue', () => {
  let wrapper

  beforeEach(() => {
    window.cookieStore = {
      get: jest.fn(),
      set: jest.fn(),
    }
    navigator.vibrate = jest.fn()
    wrapper = shallowMount(GamePage, {
      propsData: {
        difficulty: 'Easy',
      },
      stubs: {
        Confetti: true,
      },
      mocks: {
        $store: {
          state: {
            user: {
              points: 0,
              name: 'Test User',
              difficulty: 'Easy',
            },
          },
        },
      },
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
    jest.clearAllTimers()
  })

  it('starts the game when created', () => {
    expect(wrapper.vm.gameInterval).not.toBeNull()
  })

  it('ends the game when time is up', () => {
    jest.advanceTimersByTime(60000)
    expect(wrapper.vm.isEndedGame).toBe(true)
  })

  it('restarts the game', () => {
    wrapper.vm.restartGame()
    expect(wrapper.vm.isEndedGame).toBe(false)
    expect(wrapper.vm.timeLeft).toBe(60)
  })

  it('moves the mole', () => {
    const initialMoleIndex = wrapper.vm.moleIndex
    wrapper.vm.moveMole()
    expect(wrapper.vm.moleIndex).not.toBe(initialMoleIndex)
  })

  it('handles click correctly when mole is hit', () => {
    const moleIndex = wrapper.vm.moleIndex
    wrapper.vm.handleClick(moleIndex)
    expect(wrapper.vm.points).toBe(10)
  })

  it('handles click correctly when mole is not hit', () => {
    const moleIndex = (wrapper.vm.moleIndex + 1) % 9
    wrapper.vm.handleClick(moleIndex)
    expect(wrapper.vm.points).toBe(0)
  })
})
