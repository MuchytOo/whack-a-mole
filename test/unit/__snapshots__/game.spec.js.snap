// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`game.vue renders correctly 1`] = `
<!--function (el) {
      var obj;

      var args = [], len = arguments.length - 1;
      while ( len-- &gt; 0 ) args[ len ] = arguments[ len + 1 ];
      if (shouldNotBeStubbed(el, stubs, modifiedComponents)) {
        return originalCreateElement.apply(void 0, [ el ].concat( args ))
      }

      if (isConstructor(el) || isComponentOptions(el)) {
        var componentOptions = isConstructor(el) ? el.options : el;
        var elName = componentOptions.name;

        var stubbedComponent = resolveComponent(elName, stubs);
        if (stubbedComponent) {
          return originalCreateElement.apply(void 0, [ stubbedComponent ].concat( args ))
        }

        if (stubAllComponents) {
          var stub = createStubFromComponent(el, elName || 'anonymous', _Vue);
          return originalCreateElement.apply(void 0, [ stub ].concat( args ))
        }
        var Constructor = shouldExtend(el) ? extend(el, _Vue) : el;

        return originalCreateElement.apply(void 0, [ Constructor ].concat( args ))
      }

      if (typeof el === 'string') {
        var original = resolveComponent(el, originalComponents);

        if (!original) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        if (isDynamicComponent(original)) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        var stub$1 = createStubIfNeeded(stubAllComponents, original, _Vue, el);

        if (stub$1) {
          Object.assign(vm.$options.components, ( obj = {}, obj[el] = stub$1, obj ));
          modifiedComponents.add(el);
        }
      }

      return originalCreateElement.apply(void 0, [ el ].concat( args ))
    }-->
`;
